java-spring6/src/main/java/nz/co/blink/debit/client/v1/BlinkDebitClient.java

```diff
@@ -1599,14 +1566,13 @@ public Mono<Payment> awaitSuccessfulPaymentAsMono(UUID paymentId, final int maxW
                         case ACCEPTEDSETTLEMENTCOMPLETED:
                             break;
                         case REJECTED:
-                            BlinkPaymentRejectedException exception1 =
-                                    new BlinkPaymentRejectedException("Payment [" + paymentId
+                            BlinkServiceException exception1 =
```

Comment:
It might be useful here to have a payment rejected exception so that clients could tell programatically that the payment was rejected, rather than the generic service exception